src/content/app/species-selector/views/species-manager/selected-genomes-table/SelectedGenomesTable.tsx

```diff
@@ -14,10 +14,346 @@
  * limitations under the License.
  */
 
-import React from 'react';
+import React, { Fragment, useReducer } from 'react';
+import { Link } from 'react-router-dom';
+
+import { useAppSelector, useAppDispatch } from 'src/store';
+
+import * as urlFor from 'src/shared/helpers/urlHelper';
+
+import { getCommittedSpecies } from 'src/content/app/species-selector/state/species-selector-general-slice/speciesSelectorGeneralSelectors';
+
+import { deleteSpeciesAndSave } from 'src/content/app/species-selector/state/species-selector-general-slice/speciesSelectorGeneralSlice';
+import { toggleSpeciesUseAndSave } from 'src/content/app/species-selector/state/species-selector-general-slice/speciesSelectorGeneralSlice';
+
+import { Table, ColumnHead } from 'src/shared/components/table';
+
+import DeleteButton from 'src/shared/components/delete-button/DeleteButton';
+
+import { PrimaryButton } from 'src/shared/components/button/Button';
+import Checkbox from 'src/shared/components/checkbox/Checkbox';
+import SlideToggle from 'src/shared/components/slide-toggle/SlideToggle';
+import {
+  CommonName,
+  ScientificName,
+  AssemblyName,
+  AssemblyAccessionId,
+  SpeciesType
+} from 'src/shared/components/species-name-parts-for-table';
+
+import HomeIcon from 'static/icons/icon_home.svg';
+
+import type { CommittedItem } from 'src/content/app/species-selector/types/committedItem';
+
+import styles from './SelectedGenomesTable.module.css';
+
+/**
+ * TODO:
+ * - Respond to screen width (show more / show less)
+ * - When the last species is deleted, navigate (where? just one step back?)
+ *
+ * Qs:
+ * - What to do if there are no selected species?
+ *   Example: delete species, get transported to the species selector home page,
+ *   then go back one page in browser history?
+ */
+
+type DeletionModeSettings = {
+  rowDeletionInitiated: string;
+  genomeIds: string[];
+};
+
+type TableState = {
+  isShowingAllColumns: boolean;
+  deletionModeSettings: DeletionModeSettings | null;
+};
+
+type EnterDeletionModeAction = {
+  type: 'enter-deletion-mode';
+  genomeId: string;
+};
+
+type ExitDeletionModeAction = {
+  type: 'exit-deletion-mode';
+};
+
+type UpdateDeletionListAction = {
+  type: 'update-deletion-list';
+  genomeIds: string[];
+};
+
+type DispatchedTableAction =
+  | EnterDeletionModeAction
+  | ExitDeletionModeAction
+  | UpdateDeletionListAction;
+
+// The function accepts id of the genome in the table row where the delete button was clicked
+const getNewDeleteModeSettings = (genomeId: string): DeletionModeSettings => {
+  return {
+    rowDeletionInitiated: genomeId,
```

Comment:
This key name seems like it wants a boolean so is a bit confusing as to what it is doing

Reply:
It's not a boolean though. It is a string that keeps track of which row in the table contains the Delete button that was clicked to toggle the genome removal mode on.

Are you suggesting that this field should be named differently? Could you suggest a name?

Reply:
The message I wanted to convey is "genome id associated with the row that triggered the deletion mode". In `genomeRowForDeletion`, I am reading "the genome row that is marked for deletion".

Reply:
Okay so how about `genomeTriggeringDeletion`?

Reply:
`initialGenomeId`?

Reply:
That is okay too